pages/blog/generated-columns-in-crate.html

```diff
@@ -0,0 +1,187 @@
+title: Generated Columns in Crate.IO 0.54.0
+link: https://crate.io/blog/generated-columns-in-crate/
+author: Ruslan Kovalov
+description: New Crate.IO release with the new ``Generated Columns`` feature is out!
+created: 2015-12-01
+post_name: generated-columns-in-crate
+status: publish
+post_type: post
+tags: generated columns, generation expression
+category: developernews
+
+_December 01, 2015 – Dornbirn, Austria_
+
+
+We are pleased to announce the support of `generated columns` in **Crate.io**. The new functionality will be available in the fresh **0.54.0** testing release!
+
+The idea behind *generated columns* is to have its values computed from *generation expressions* provided at the time when the *columns* are defined, unlike regular columns, which values are set explicitly.
+
+*Generated columns* provide us with flexibility in defining *generation expressions*, where it is also possible to create complex expressions as well as reference values of other columns.
+Providing a data type for such columns is optional. It is inferred by the return type of the supplied expression.
+
+*Crate.io* supports only *virtual generated columns* which are the read-only and computed internally columns.
+
+
+## Use cases
+
+With *Generated columns* we get an effortless way to solve a variety of problems. Let us take a closer look at couple of them:
+
+### Partition by expression
+
+Imagine that you have to analyze time-based data, e.g. a log of enormous size, where only records containing in particular time slices would have a value for you. The problem you have encountered here is that the queries you run against the log, due to the size and nature of data, are way to slow. You will definitely try to come up with a smart solution to improve the query speed. One convenient way to accomplish that is to look up for data in those sliced data samples which contain the records of your interest, instead of iterating over all existing records. That is where [partitioned tables](https://crate.io/docs/en/latest/sql/partitioned_tables.html) come to the rescue. Having this functionality let you have separate partitions but still query them like a single table.
```

Comment:
try to simplify the structure of the sentences in this paragraph. maybe @ChrisChinchilla can help?